### ISSUE TITLE
Runtime exception thrown {"Class not registered (0x80040154 (REGDB_E_CLASSNOTREG))"}

### ISSUE THREAD

**Issue by Rochishaa** [microsoft/CsWinRT#647]
I am getting the following runtime error when consuming the interop assembly. My understanding was that winrt.runtime.dll will take care of registering the classes. Do we still need to manually add them to the app.manifest ?

![2021-01-04_18-37-15](https://user-images.githubusercontent.com/71650482/103538780-e3f16b80-4ebc-11eb-886b-54da50e06ded.png)

◢ | $exception | {"Exception has been thrown by the target of an invocation."} | System.Reflection.TargetInvocationException

▶ | InnerException | {"Class not registered (0x80040154 (REGDB_E_CLASSNOTREG))"} | System.Exception {System.Runtime.InteropServices.COMException}

  | Message | "Exception has been thrown by the target of an invocation." | string

  | StackTrace | "   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& hasNoDefaultCtor)\r\n   at System.RuntimeType.CreateInstanceDefaultCtorSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean fillCache)\r\n   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)\r\n   at System.Activator.CreateInstance[T]()\r\n   at WinRT.WeakLazy`1.get_Value()\r\n   at WinRT.ActivationFactory`1.ActivateInstance[I]()\r\n   at SimpleMathComponent.SimpleMath..ctor()\r\n   at ConsoleAppSample.Program.Main(String[] args) in C:\\Dev\\CsWinRt\\src\\Samples\\Net5ProjectionSample\\ConsoleAppSample\\Program.cs:line 9" | string

**Comment by Scottj1s** [comment]
winrt.runtime.dll provides activation fallback logic (see **WinRT type activation** [here](https://docs.microsoft.com/en-us/windows/uwp/csharp-winrt/#reference-an-interop-assembly)), based on naming conventions.  But ideally, you should still manually register any classes you want to activate, as CsWinRT cannot know about custom (non-conventional) mappings of class name to host DLL name and/or path.

**Comment by Scottj1s** [comment]
Closing issue - please reopen if activation issues remain after registration

**Comment by Rochishaa** [comment]
Hello Scott, Thank you for the prompt response.

In my current UWP project, I was using C++/WinRT components. The registration there was done automatically if I added the reference to the .winmd. I am looking at migrating the same app to WinUI 3.

For starters, I am trying to consume SimpleMathComponent C++/WinRT component in a sample WinUI 3.0 Preview 3 Desktop app. Following the steps as described [here](https://docs.microsoft.com/en-us/windows/uwp/csharp-winrt/net-projection-from-cppwinrt-component), I created .NET 5.0 projection and added the nuget package as reference to my sample winui for desktop app. Build is successful, but still getting runtime exception. The reason being missing registration. 

So, after the clarification you provided, I now manually added registration to the app.manifest in WinUI sample app. 

This is what my WinUI app.manifest looks like : 

`<?xml version="1.0" encoding="utf-8"?>
<assembly manifestVersion="1.0" xmlns="urn:schemas-microsoft-com:asm.v1">
  <assemblyIdentity version="1.0.0.0" name="App1.app"/>
  <file name="SimpleMathProjection.dll">
    <activatableClass
    name="SimpleMathComponent.SimpleMath"
    threadingModel="both"
    xmlns="urn:schemas-microsoft-com:winrt.v1" />
  </file>
  <application xmlns="urn:schemas-microsoft-com:asm.v3">
    <windowsSettings>
      
      <dpiAware xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">true/PM</dpiAware>
      <dpiAwareness xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">PerMonitorV2, PerMonitor</dpiAwareness>
    </windowsSettings>
  </application>
</assembly>`

This is the exception I am getting this time around:

System.Reflection.TargetInvocationException
  HResult=0x80131604
  Message=Exception has been thrown by the target of an invocation.
  Source=System.Private.CoreLib
  StackTrace:
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& hasNoDefaultCtor)
   at System.RuntimeType.CreateInstanceDefaultCtorSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean fillCache)
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)
   at System.Activator.CreateInstance[T]()
   at WinRT.WeakLazy`1.get_Value()
   at WinRT.ActivationFactory`1.ActivateInstance[I]()
   at SimpleMathComponent.SimpleMath..ctor()
   at App1.MainWindow.myButton_Click(Object sender, RoutedEventArgs e) in C:\Dev\CsWinRt\src\Samples\Net5ProjectionSample\App1\App1\MainWindow.xaml.cs:line 35
   at ABI.Microsoft.UI.Xaml.RoutedEventHandler.<>c__DisplayClass10_0.<Do_Abi_Invoke>b__0(RoutedEventHandler invoke)
   at WinRT.ComWrappersSupport.MarshalDelegateInvoke[T](IntPtr thisPtr, Action`1 invoke)
   at ABI.Microsoft.UI.Xaml.RoutedEventHandler.Do_Abi_Invoke(IntPtr thisPtr, IntPtr sender, IntPtr e)

  This exception was originally thrown at this call stack:
    [External Code]

Inner Exception 1:
COMException: Error in the DLL (0x800401F9 (CO_E_ERRORINDLL))
